File: app-core/src/main/java/ru/micron/dto/PageableData.java
package ru.micron.dto;

import javax.validation.ValidationException;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class PageableData {

  private long totalCount;
  private long totalPages;
  private long pageNum;
  private long pageSize;
  private long currentPageSize;

  public PageableData(long totalCount, long pageNum, long pageSize, long currentPageSize) {
    validateMinValue(pageNum, 1);
    validateMinValue(pageSize, 1);
    validateMinValue(totalCount, 0);

    this.totalCount = totalCount;
    this.pageNum = pageNum;
    this.pageSize = pageSize;
    this.totalPages =
        totalCount % pageSize == 0 ? totalCount / pageSize : totalCount / pageSize + 1;
    this.currentPageSize = currentPageSize;
  }

  private void validateMinValue(long value, int minValue) {
    if (value < minValue) {
      throw new ValidationException(
          String.format("The current value [%s] is less than min possible value [%s] ", value,
              minValue));
    }
  }

}

File: app-core/src/main/kotlin/ru/micron/dto/PageableData.kt
package ru.micron.dto

import javax.validation.ValidationException

data class PageableData(
    var totalCount: Long,
    var pageNum: Long,
    var pageSize: Long,
    var currentPageSize: Long
) {

    var totalPages: Long

    init {
        validateMinValue(pageNum, 1)
        validateMinValue(pageSize, 1)
        validateMinValue(totalCount, 0)
        totalPages =
            if (totalCount % pageSize == 0L)
                totalCount / pageSize
            else
                totalCount / pageSize + 1
    }

    private fun validateMinValue(value: Long, minValue: Int) {
        if (value < minValue) {
            throw ValidationException(
                String.format(
                    "The current value [%s] is less than min possible value [%s] ", value,
                    minValue
                )
            )
        }
    }
}